dm/dm/worker/source_worker.go

```diff
@@ -787,7 +800,9 @@ func (w *SourceWorker) operateSubTaskStage(stage ha.Stage, subTaskCfg config.Sub
 		if st := w.subTaskHolder.findSubTask(stage.Task); st == nil {
 			// create the subtask for expected running and paused stage.
 			log.L().Info("start to create subtask", zap.String("sourceID", subTaskCfg.SourceID), zap.String("task", subTaskCfg.Name))
-			err := w.StartSubTask(&subTaskCfg, stage.Expect, true)
+			// new subtask, pass Running to validator stage, let internal logic determines it.
+			// as there's no new subtask with a stopped validator, either it has a Running one or none
+			err := w.StartSubTask(&subTaskCfg, stage.Expect, pb.Stage_Running, true)
```

Comment:
always `StartSubTask` with  expected running validator is confused me, if user config validator in task config with all mode, will the validator start automatically when task into incr sync satge ?

but what happened if user start task with full mode?

Reply:
It does look confusing. But the validator won't start even passing the `pb.Stage_Running` flag as your previous comment points out...Every time starting the validator, the validator's `mode` will be checked first and forced to return if it's neither `full` and `fast`.
In other words, the validator can still be started in `all` mode migration, where binlog should not be migrated, as long as the mode is set to `full` or `fast`.

Reply:
But I am also confused by this parameter. It seems always `Running`.

Reply:
i explained it in the comment, it's ok to set expect `stage=running` for new subtask, but you're right, it's not explicit and confusing.